tsl/src/compression/decompress_text_test_impl.c

```diff
@@ -101,6 +101,11 @@ decompress_generic_text(const uint8 *Data, size_t Size, bool bulk, int requested
 	int nn = 0;
 	for (DecompressResult r = iter->try_next(iter); !r.is_done; r = iter->try_next(iter))
 	{
+		if (nn >= n)
+		{
+			elog(ERROR, "the repeated recompression result doesn't match");
+		}
+
 		if (r.is_null != results[nn].is_null)
```

Comment:
The check is different from the one last in the loop and there is no increment in the for-loop that changes `nn`. Is this correct?

Reply:
The increment is at the end of the loop. The reason to move this check to the front is so that it works properly for n = 0 as well.